src/Decoder.c

```diff
@@ -3792,6 +3922,39 @@ static ZyanStatus ZydisNodeHandlerOpcode(ZydisDecoderState* state,
         default:
             ZYAN_UNREACHABLE;
         }
+
+        if (instruction->encoding != ZYDIS_INSTRUCTION_ENCODING_REX2)
+        {
+            break;
+        }
+
+        ZyanBool illegal_rex2 = (instruction->opcode == 0x0F);
```

Comment:
Might be better to just generate a flag for this in the generator based on the same condition. We usually try to not hardcode table related conditions.

Reply:
There are as well XSAVE and XRSTORE variants that do not support REX2.